### PR TITLE
Rename regions to view everywhere

### PR DESCRIPTION
**Pull request by Phlya**
I left the old names in places where we need to fix the code anyway, e.g. asym expected

### CODE REVIEW

File: cooltools/expected.py [hunk 1]
```diff
@@ -557,8 +557,8 @@ def diagsum(
     ----------
     clr : cooler.Cooler
         Cooler object
-    regions : viewframe or sequence of genomic range tuples
-        Support regions for intra-chromosomal diagonal summation
+    view_df : viewframe or sequence of genomic range tuples
```
Comment (agalitsyna) [inline cooltools/expected.py:560]: Looks like diagsum does not support tuples anymore

File: cooltools/saddle.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ def digitize_track(binedges, track, regions=None):
         edges. See encoding details in Notes.
     track : tuple of (DataFrame, str)
         bedGraph-like dataframe along with the name of the value column.
-    regions: viewframe, sequence of str or tuples
+    view_df: viewframe, sequence of str or tuples
```
Comment (agalitsyna) [inline cooltools/saddle.py:76]: No backward compatibility: looks like view_df cannot be a sequence of str or tuples anymore

File: cooltools/saddle.py [hunk 3]
```diff
@@ -283,7 +286,7 @@ def make_saddle(
     contact_type : str
         If 'cis' then only cis interactions are used to build the matrix.
         If 'trans', only trans interactions are used.
-    regions : viewframe, sequence of str or tuple, optional
+    view_df : viewframe, sequence of str or tuple, optional
```
Comment (agalitsyna) [inline cooltools/saddle.py:289]: Looks like view_df cannot be a sequence of str or tuple anymore?

**Review by agalitsyna — CHANGES_REQUESTED** [review]
I just noticed (unrelated to the work that Ilya has done) that at least three functions have lost backward compatibility with old code. For example, if someone uses old code with diagsum, it won't support sequences of tuples as input. 
I would retain backward compatibility of diagsum/digitize_track/etc. If we don't want it, the docstrings are to be changed.

**Comment by Phlya** [comment]
I thought that was the decision, that we require a proper viewframe from the user now? https://github.com/open2c/cooltools/issues/254

**Comment by sergpolly** [comment]
@Phlya that what I was wondering, but @nvictus suggested that it should be phased out (in) gradually at least ...

I wonder if we could replace those `assert` with something like this:
```python
try:
     bioframe.is_viewframe(view_df, ...,raise_errors=True)
     bioframe.is_contained(...,raise_errors=True)
except WhatEverError as e:
    # https://docs.python.org/3/library/warnings.html#warnings.warn
    warning.warn("view_df has to be a proper viewframe from next release on ...", DeprecationWarning, stacklevel=2)
    # try to make viewframe from whatever was provided
    view_df = bioframe.make_viewframe(view_df)
```

**Pull request merged**